src/Database/DriverInterface.php

```diff
@@ -0,0 +1,260 @@
+     * transformed to accommodate any specificities of the SQL dialect in use.
+     *
+     * @param string $type The type of query to be transformed
+     * (select, insert, update, delete).
+     * @return callable
+     */
+    public function queryTranslator($type);
+
+    /**
+     * Get the schema dialect.
+     *
+     * Used by Cake\Database\Schema package to reflect schema and
+     * generate schema.
+     *
+     * If all the tables that use this Driver specify their
+     * own schemas, then this may return null.
+     *
+     * @return \Cake\Database\Schema\BaseSchema
+     */
+    public function schemaDialect();
+
+    /**
+     * Quotes a database identifier (a column name, table name, etc..) to
+     * be used safely in queries without the risk of using reserved words.
+     *
+     * @param string $identifier The identifier expression to quote.
+     * @return string
+     */
+    public function quoteIdentifier($identifier);
+
+    /**
+     * Escapes values for use in schema definitions.
+     *
+     * @param mixed $value The value to escape.
+     * @return string String for use in schema definitions.
+     */
+    public function schemaValue($value);
+
+    /**
+     * Returns the schema name that's being used.
+     *
+     * @return string
+     */
+    public function schema();
+
+    /**
+     * Returns last id generated for a table or sequence in database.
+     *
+     * @param string|null $table table name or sequence to get last insert value from.
+     * @param string|null $column the name of the column representing the primary key.
+     * @return string|int
+     */
+    public function lastInsertId($table = null, $column = null);
+
+    /**
+     * Checks whether or not the driver is connected.
+     *
+     * @return bool
+     */
+    public function isConnected();
+
+    /**
+     * Sets whether or not this driver should automatically quote identifiers
+     * in queries.
+     *
+     * @param bool $enable Whether to enable auto quoting
+     * @return $this
+     */
+    public function enableAutoQuoting($enable = true);
+
+    /**
+     * Returns whether or not this driver should automatically quote identifiers
+     * in queries.
+     *
+     * @return bool
+     */
+    public function isAutoQuotingEnabled();
+
+    /**
+     * Transforms the passed query to this Driver's dialect and returns an instance
+     * of the transformed query and the full compiled SQL string.
+     *
+     * @param \Cake\Database\Query $query The query to compile.
+     * @param \Cake\Database\ValueBinder $generator The value binder to use.
+     * @return array containing 2 entries. The first entity is the transformed query
+     * and the second one the compiled SQL.
+     */
+    public function compileQuery(Query $query, ValueBinder $generator);
+
+    /**
+     * Returns an instance of a QueryCompiler.
+     *
+     * @return \Cake\Database\QueryCompiler
+     */
+    public function newCompiler();
```

Comment:
I would like getCompiler() more than newCompiler(), as for a normal internal factory call.

Reply:
Yeah this method is in use already.